### PR TITLE
Add AppConfiguration Builder

### PR DESCRIPTION
**Pull request by rorbech**

### CODE REVIEW

File: packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt [hunk 1]
```diff
@@ -16,37 +16,74 @@
...
 
+import io.realm.internal.platform.singleThreadDispatcher
 import io.realm.mongodb.internal.AppConfigurationImpl
 import kotlinx.coroutines.CoroutineDispatcher
 
 /**
- * TODO
+ * An **AppConfiguration** is used to setup linkage to a MongoDB Realm application.
+ *
+ * Instances of a AppConfiguration can only created by using the
```
Comment (cmelchior) [inline packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt:26]:
```suggestion
 * Instances of a AppConfiguration can only be created by using the
```

File: packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt [hunk 2]
```diff
@@ -16,37 +16,74 @@
...
-    appId: String,
-    baseUrl: String,
-    dispatcher: CoroutineDispatcher
-): AppConfiguration {
-    return AppConfigurationImpl(appId, baseUrl, dispatcher)
+    /**
+     * Builder used to construct instances of a {@link AppConfiguration} in a fluent manner.
+     */
+    class Builder(
+        val appId: String
```
Comment (cmelchior) [inline packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt:60]: Missing Dokka for this

**Review by cmelchior — APPROVED** [review]
Nice 💯

File: packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt [hunk 3]
```diff
@@ -16,37 +16,74 @@
...
-// FIXME Initialize with proper multithreaded dispatcher
-//  https://github.com/realm/realm-kotlin/issues/450
-fun appConfigurationOf(
-    appId: String,
-    baseUrl: String,
-    dispatcher: CoroutineDispatcher
-): AppConfiguration {
-    return AppConfigurationImpl(appId, baseUrl, dispatcher)
+    /**
+     * Builder used to construct instances of a {@link AppConfiguration} in a fluent manner.
```
Comment (edualonso) [inline packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt:57]: The format you used is Javadoc, not dokka.
```suggestion
     * Builder used to construct instances of an [AppConfiguration] in a fluent manner.
```

**Review by edualonso — APPROVED** [review]
Looking good 😅 Just make sure the docs are Dokka-formatted and not Javadoc.

**Pull request merged**